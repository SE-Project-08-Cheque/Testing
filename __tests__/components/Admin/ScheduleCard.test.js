// import dependencies
import React from "react";

// import react-testing methods
import { render, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import ScheduleCard from "../../../components/Admin/ScheduleCard";


afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ScheduleCard path={[]} />, div);
});

it("renders schedule card correctly", () => {
  var sample_path = ['mirissa', 'galle'];
  const { getByTestId, getAllByTestId } = render(
    <ScheduleCard
      user = "Chandima"
      fair = {100}
      path = {sample_path}
      travelmethod = "Bus"
    />
  );
  expect(getByTestId('fair')).toHaveTextContent("Rs. 100");
  expect(getByTestId('user')).toHaveTextContent("Chandima");
  expect(getByTestId('method')).toHaveTextContent("Bus");
  getAllByTestId('path').forEach((value, index)=>{
    expect(value).toHaveTextContent(sample_path[index]);
  })
  

});

it("matches snapshot", () => {
  const tree = renderer.create(<ScheduleCard
    user = "Chandima"
    fair = {100}
    path = {['mirissa', 'galle']}
    travelmethod = "Bus"
  />);

    expect(tree).toMatchSnapshot();
})

