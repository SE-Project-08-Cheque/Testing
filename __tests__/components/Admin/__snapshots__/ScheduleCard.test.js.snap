// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-136pjoj"
>
  <div
    className="chakra-stack css-c0d8a5"
  >
    <span
      className="chakra-badge css-p5nc7k"
    />
    <span
      className="chakra-badge css-166rvjf"
    />
  </div>
  <div
    className="css-61ehcq"
  >
    <div
      className="css-0"
    >
      <p
        className="chakra-text css-1lj023j"
      >
         Schedule By,
      </p>
      <p
        className="chakra-text css-1lj023j"
        data-testid="user"
      >
        <span
          className="chakra-avatar css-1ohlb5u"
        >
          <div
            aria-label="user"
            className="chakra-avatar__initials css-0"
          >
            u
          </div>
        </span>
        Chandima
      </p>
    </div>
    <div
      className="css-0"
    >
      <p
        className="chakra-text css-1lj023j"
      >
        Fair
      </p>
      <p
        className="chakra-text css-6dvxm2"
        data-testid="fair"
      >
        Rs. 
        100
      </p>
    </div>
  </div>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
    role="separator"
  />
  <p
    className="chakra-text css-8j1nyb"
  >
    Path
  </p>
  <div
    className="chakra-stack css-19kpmhn"
  >
    <span
      className="chakra-badge css-ifsvay"
      data-testid="path"
    >
      mirissa
    </span>
    <span
      className="chakra-badge css-ifsvay"
      data-testid="path"
    >
      galle
    </span>
  </div>
  <div
    className="css-61ehcq"
  >
    <div
      className="css-0"
    >
      <p
        className="chakra-text css-8j1nyb"
      >
        Travel Method
      </p>
      <span
        className="chakra-badge css-ifsvay"
        data-testid="method"
      >
        Bus
      </span>
    </div>
    <div
      className="css-0"
    >
      <p
        className="chakra-text css-8j1nyb"
      >
        Date
      </p>
      <span
        className="chakra-badge css-ifsvay"
      />
    </div>
  </div>
</div>
`;
